src/utils/upload/fileUploadUtils.js

```diff
@@ -247,30 +253,74 @@ const fileUploadUtils = {
         return { valid: validFiles, invalid: invalidFiles };
       };
 
-      const { valid: DGEUnfilteredFiles, invalid } = getFilesMatching('DGE_unfiltered', files);
-      const { valid: DGEFilteredFiles } = getFilesMatching('DGE_filtered', invalid);
+      const dgeUnfilteredFiles = getFilesMatching('DGE_unfiltered/', files);
+      const dgeFilteredFiles = getFilesMatching('DGE_filtered/', dgeUnfilteredFiles.invalid);
+      const noMiddlePathFiles = getFilesMatching('', dgeFilteredFiles.invalid);
 
-      const filesToUpload = _.uniq([...Object.keys(DGEFilteredFiles), ...Object.keys(DGEUnfilteredFiles)])
+      // These are the ones that didn't match any of the 3 accepted shapes
+      const invalidFiles = noMiddlePathFiles.invalid.map((file) => ({
+        path: file.path,
+        rejectReason: 'Invalid file path. It should be in the form "sample/DGE_unfiltered/file", "sample/DGE_filtered/file" or "sample/file".',
+      }));
+
+      const filesToUpload = _.uniq([
+        ...Object.entries(dgeFilteredFiles.valid),
+        ...Object.entries(dgeUnfilteredFiles.valid),
+        ...Object.entries(noMiddlePathFiles.valid)])
         // Only allow sample-specific files, not all samples in one files
-        .filter((sampleName) => !['all-sample', 'all-well', 'All Wells'].includes(sampleName))
-        // if unfiltered files are present, pick them. Otherwise, pick the filtered files
-        .flatMap((sampleName) => (
-          DGEUnfilteredFiles[sampleName] ?? DGEFilteredFiles[sampleName]
+        .filter(([sampleName, sampleFiles]) => {
+          const accepted = !['all-sample', 'all-well', 'All Wells'].includes(sampleName);
```

Comment:
to make this easier to read I suggest having a variable for not accepted and removing the `!`, so like this:
```
const notAccepted = ['all-sample', 'all-well', 'All Wells'].includes(sampleName);
```

Reply:
oh I remembered why I did this, becuase I would need to do `return !notAccepted` in the end of that function

Reply:
yes, let's leave it like this. Double negation sounds worse